scripts/quantitative_testing.py

```diff
@@ -0,0 +1,122 @@
+    BLOCK: "Usage: `block <num>`",
+    END_TURN: "End the turn. Usage: `endturn`",
+    RESET_TURNS: "Reset the turn counter. Usage: `resetturns`",
+    SHOW: "Show the current results. Usage: `show`",
+    TEAM: "Set the current team. Usage `team Aborah pythia entropy",
+    HELP: "Display the usage for a command. Usage: `help <command>",
+}
+ACTIONS = list(HELP_STR.keys())
+COMPANIONS = ["Pythia", "Aborah", "Architect", "Clown", "Entropy"]
+
+
+def closest_word(word: str, words: List[str]) -> str:
+    edit_distances = [nltk.edit_distance(word, w) for w in words]
+    for i, _ in enumerate(edit_distances):
+        if words[i].lower()[:2] == word.lower()[:2]:
+            edit_distances[i] -= 5
+    min_distance = min(edit_distances)
+    chosen = words[edit_distances.index(min_distance)]
+    return chosen
+
+
+def get_cleaned_input() -> List[str]:
+    raw = input()
+    sep = [s.strip().lower() for s in raw.split(" ")]
+    return [s for s in sep if len(s) > 0]
+
+
+def main():
+    print(f"Available actions: {', '.join(ACTIONS)}")
+
+    print("Please enter starting list of companions: ", end="")
+    sep_companions = get_cleaned_input()
+    selected_companions = [closest_word(c, COMPANIONS) for c in sep_companions]
+    print("Selected companions:", selected_companions)
+
+    turn_counter = 1
+    current_turn_data = {
+        "turn": turn_counter,
+        "block": 0,
+        "damage": 0,
+        "companions": selected_companions[:],
+    }
+    turns = []
+
+    while True:
+        print("> ", end="")
+        split = get_cleaned_input()
+        raw_action = split[0]
+
+        chosen_action = closest_word(raw_action, ACTIONS)
+        print(f"Chosen action: {chosen_action}")
+
+        if chosen_action == DAMAGE:
+            if len(split) < 2:
+                print(HELP_STR[DAMAGE])
+                continue
+
+            num = int(split[1])
+            current_turn_data["damage"] += num
+        elif chosen_action == BLOCK:
+            if len(split) < 2:
+                print(HELP_STR[BLOCK])
+                continue
+
+            num = int(split[1])
+            current_turn_data["block"] += num
+        elif chosen_action == END_TURN:
+            current_turn_data["turn"] = turn_counter
+            print("Totals")
+            print(current_turn_data)
+            turns.append(current_turn_data)
+
+            turn_counter += 1
+            current_turn_data = {
+                "turn": turn_counter,
+                "block": 0,
+                "damage": 0,
+                "companions": selected_companions[:],
+            }
+        elif chosen_action == RESET_TURNS:
+            turn_counter = 1
+            current_turn_data["turn"] = turn_counter
+        elif chosen_action == SHOW:
+            df = pd.DataFrame(turns)
+            print("Cumulative data")
+            print(df)
+
+            print("Current turn data")
+            print(current_turn_data)
+        elif chosen_action == TEAM:
```

Comment:
Interesting being able to change this mid-turn-series. We're picturing this series of turns is just one combat right? 

Have some postulations about why we might have this here, for future work or other convenience, but I'll just stand by the question for now and let ya answer.

Reply:
you make a good point. I figured it would be nice to have a single interactive session across combats with the same team. Once I test using this thing, I imagine there will be some refactoring to do